### PR TITLE
Fixed evaluation for types

### LINKED ISSUE #32811: Evalute Selection does not insert the correct type for evaluated values.
**Godot version:**
3.2 alpha 2

**OS/device including version:**
Windows 10 64bit

**Issue description:**
When using evaluate selection on anything other than a number it does not insert the type into the resulting evaluated value. For example `Vector2(1, 1) + Vector2(2, 2)` becomes `(3, 3)` when it should be `Vector2(3, 3)`. This happens to everything that isn't a plain number. Sometimes it doesn't even include parentheses, which is the case with Color types.

Is this an issue with the Expression class since it is being used to evaluate lines?

**Steps to reproduce:**
Type in any expression into the script editor that uses types other than numbers, then evaluate that line. Here are a few examples:

```GDScript
Vector2(1, 1) + Vector2(2, 2) # becomes (3, 3)
Vector3(1, 1, 1).normalized() # becomes (0.57735, 0.57735, 0.57735)
Color(1, 1, 1, 1).darkened(0.5) # becomes 0.5,0.5,0.5,1
````

### PR DESCRIPTION
**Pull request by DavidSichma**
fixes #32811

### CODE REVIEW

File: editor/plugins/script_text_editor.cpp [hunk 1]
```diff
@@ -1186,7 +1186,7 @@ void ScriptTextEditor::_edit_option(int p_op) {
 				if (expression.parse(line) == OK) {
 					Variant result = expression.execute(Array(), Variant(), false);
```
Comment (groud) [inline editor/plugins/script_text_editor.cpp]: You should remove this variable then.

Comment (KoBeWi) [reply editor/plugins/script_text_editor.cpp]: But it's still used.

Comment (groud) [reply editor/plugins/script_text_editor.cpp]: Ah my bad, I misread the remaining of the line. Everything looks good then. :)

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**